From fba948b39f6e3b2e70caa1880903f8fb17faf6d9 Mon Sep 17 00:00:00 2001
From: Xaver Hugl <xaver.hugl@gmail.com>
Date: Fri, 19 Apr 2024 23:06:56 +0200
Subject: [PATCH] scene/workspacescene: don't check direct scanout candidates
 for a pixmap

We don't need a pixmap for direct scanout, and the drm backend destroys the pixmap
when direct scanout is successful... so this check created a loop of direct scanout
working and not working, and worse, the client reallocating its buffers each time.

BUG: 485639
BUG: 485730
BUG: 485712
CCBUG: 477016
---
 src/scene/workspacescene.cpp | 12 +-----------
 1 file changed, 1 insertion(+), 11 deletions(-)

diff --git a/src/scene/workspacescene.cpp b/src/scene/workspacescene.cpp
index d099a7321d3..d2440bb3e3d 100644
--- a/src/scene/workspacescene.cpp
+++ b/src/scene/workspacescene.cpp
@@ -164,17 +164,7 @@ static SurfaceItem *findTopMostSurface(SurfaceItem *item)
             }
         }
     }
-    if (item->pixmap()) {
-        return item;
-    }
-    for (const auto &child : children | std::views::reverse) {
-        if (child->z() < 0) {
-            if (auto item = findTopMostSurface(static_cast<SurfaceItem *>(child))) {
-                return item;
-            }
-        }
-    }
-    return nullptr;
+    return item;
 }
 
 SurfaceItem *WorkspaceScene::scanoutCandidate() const
-- 
GitLab

